src/ex_cmds2.c

```diff
@@ -457,6 +457,31 @@ ex_listdo(exarg_T *eap)
     tabpage_T	*tp;
     buf_T	*buf = curbuf;
     int		next_fnum = 0;
+
+    if (curwin->w_p_wfb)
+    {
+        if ((eap->cmdidx == CMD_ldo || eap->cmdidx == CMD_lfdo) && !eap->forceit)
+        {
+            // Disallow :ldo if 'winfixbuf' is applied
+            semsg("%s", e_winfixbuf_cannot_go_to_buffer);
+            return;
+        }
+
+        if (win_valid(prevwin))
+            // Change the current window to another because 'winfixbuf' is enabled
+            curwin = prevwin;
+        else
+        {
+            // Split the window, which will be 'nowinfixbuf', and set curwin to that
+            exarg_T new_eap;
+            CLEAR_FIELD(new_eap);
+            new_eap.cmdidx = CMD_split;
+            new_eap.cmd = (char_u *)"split";
+            new_eap.arg = (char_u *)"";
+            ex_splitview(&new_eap);
```

Comment:
can we have a test here?

Reply:
I could be wrong but I believe these tests already use this section

Test_argdo_make_new_window
Test_bufdo_make_new_window
Test_ldo
Test_lfdo
Test_tabdo_make_new_window

Without the split, those tests fail. Though checking again, `Test_tabdo_make_new_window` is borked and needs fixing. 

If you'd like to be absolutely certain that the split is happening as we expect, I could do what I do in `make_new_window` and count the number of window splits before and after each test, if you'd like

Reply:
If you could just fix the `Test_tabdo_make_new_window` then?

Reply:
I reread that unittest, `Test_tabdo_make_new_window` was actually fine. Though while I was in the neighborhood, I saw that `Test_normal_ctrl_w_ctrl_square_bracket_right` and `Test_normal_ctrl_w_g_ctrl_square_bracket_right` didn't match their docstrings / expected behavior and updated those.

I've also added explicit window split count tests to all unittests where it's relevant (e.g. if a window must be split, check that it is. If a window is expected to be reused, check that it is)